src/RA_AchievementSet.cpp

```diff
@@ -353,8 +350,7 @@ bool AchievementSet::LoadFromFile(ra::GameID nGameID)
         g_pCurrentGameData->SetGameID(nGameID);
                
         rapidjson::Document doc;
-        LoadDocument(doc, *pData.get());
-        if (doc.HasParseError())
+        if (LoadDocument(doc, *pData.get()) == doc.HasParseError())
```

Comment:
This logic is incorrect. `LoadDocument` may return `false` without setting `doc.HasParseError`, which would bypass the assert/return. If it makes it all the way to the end, `LoadDocument` return `!doch.HasParseError`, so you can eliminate that and just say:
```
if (!LoadDocument(doc, *pData.get()))
```

Reply:
I just saw this, I got you.